src/generic/grideditors.cpp

```diff
@@ -703,6 +707,40 @@ void wxGridCellNumberEditor::Create(wxWindow* parent,
     }
 }
 
+void wxGridCellNumberEditor::SetSize(const wxRect& r)
+{
+#if wxUSE_SPINCTRL
+    if ( HasRange() )
+    {
+        wxASSERT_MSG(m_control, "The wxSpinCtrl must be created first!");
+
+        const wxSize bestSize = Spin()->GetBestSize();
+        const int width = wxMax(r.GetWidth(), bestSize.GetWidth());
+        const int height = bestSize.GetHeight() > 0 ?
+                           bestSize.GetHeight() :
+                           r.GetHeight();
+
+        wxRect rect(r.GetLeft(), r.GetTop(), width, bestSize.GetHeight());
```

Comment:
```suggestion
        wxRect rect(r.GetPosition(), wxSize(width, height));
```

Actually I'd also use a `wxSize` instead of separate `width` and `height`, e.g.

```cpp
wxSize size = Spin()->GetBestSize();

// Extend the control to fill the entire cell horizontally.
if ( size.x < r.GetWidth() )
    size.x = r.GetWidth();

// Ensure it uses a reasonable height even if wxSpinCtrl::GetBestSize()
// didn't return anything useful.
if ( size.y <= 0 )
    size.y = r.GetHeight();
```